gradio/utils.py

```diff
@@ -528,33 +554,20 @@ def component_or_layout_class(cls_name: str) -> type[Component] | type[BlockCont
     Returns:
     cls: the component class
     """
-    import gradio.blocks
-    import gradio.components
-    import gradio.layouts
-    import gradio.templates
-
-    components = [
-        (name, cls)
-        for name, cls in gradio.components.__dict__.items()
-        if isinstance(cls, type)
-    ]
-    templates = [
-        (name, cls)
-        for name, cls in gradio.templates.__dict__.items()
-        if isinstance(cls, type)
-    ]
-    layouts = [
-        (name, cls)
-        for name, cls in gradio.layouts.__dict__.items()
-        if isinstance(cls, type)
-    ]
-    for name, cls in components + templates + layouts:
-        if name.lower() == cls_name.replace("_", "") and (
-            issubclass(cls, gradio.components.Component)
-            or issubclass(cls, gradio.blocks.BlockContext)
+    from gradio.blocks import BlockContext
+    from gradio.components import Component
+
+    components = get_all_components()
+    for cls in components:
+        if cls.__name__.lower() == cls_name.replace("_", "") and (
+            issubclass(cls, Component) or issubclass(cls, BlockContext)
         ):
             return cls
-    raise ValueError(f"No such component or layout: {cls_name}")
+    raise ValueError(
+        f"No such component or layout: {cls_name}. "
```

Comment:
I tried to automatically import the package containing the custom component but it's quite tricky. We just have a class name so in order to get the package that it's in we have to import every package installed in the system which is not ideal and cause cause things to break.